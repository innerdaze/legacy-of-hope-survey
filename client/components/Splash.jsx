import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Box from 'grommet/components/Box'
import Label from 'grommet/components/Label'

class Splash extends Component {
  render() {
    return (
      <Box
        full
        justify='center'
        align='center'
        >
        {this.props.icon}
        {this.props.logo}
        {this.props.title}
        <Label>{this.props.loadingText}</Label>
      </Box>
    )
  }
}

Splash.propTypes = {
  loadingText: PropTypes.string,
  icon: PropTypes.instanceOf(Component),
  logo: PropTypes.instanceOf(Component),
  title: PropTypes.instanceOf(Component)
}

Splash.defaultProps = {
  loadingText: 'Loading...',
  icon: null,
  logo: null,
  tite: null
}

export default Splash
