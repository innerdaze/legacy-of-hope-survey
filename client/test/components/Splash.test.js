import React from 'react'
import Splash from '../../components/Splash'

describe('Testing Splash...', () => {
  test('render Splash', () => {
    const wrapper = shallow(<Splash loadingTest='Loading...' />)

    expect(wrapper).toMatchSnapshot()
  })
})
