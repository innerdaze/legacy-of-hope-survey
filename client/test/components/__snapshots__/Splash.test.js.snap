// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Splash... render Splash 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full={true}
  justify="center"
  pad="none"
  responsive={true}
  tag="div"
>
  <Label
    size="medium"
  >
    Loading...
  </Label>
</Box>
`;
